pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -1057,7 +1079,12 @@ private void doCleanup(String broker) throws ExecutionException, InterruptedExce
 
         double cleanupTime = TimeUnit.NANOSECONDS
                 .toMillis((System.nanoTime() - startTime));
-        // TODO: clean load data stores
+
+        // clean load data stores
+        getContext().topBundleLoadDataStore().removeAsync(broker);
+        getContext().brokerLoadDataStore().removeAsync(broker);
+
+        cleanupJobs.remove(broker);
```

Comment:
no. it will be completed after this call returns.

Reply:
No, it will complete when this func returns. In fact, I think we better move this `cleanupJobs.remove(broker);` in the finally block after this call. Updated.